internal/insights/insights.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 type Service interface {
-	GetHistory(context.Context, iam.User, config.LaunchOrder) (JobHistory, error)
+	GetHistory(ctx context.Context, user iam.User, order config.LaunchOrder, source string) (JobHistory, error)
```

Comment:
How about:
```suggestion
	GetHistory(ctx context.Context, user iam.User, order config.LaunchOrder, deviceType string) (JobHistory, error)
```

Still not super clear what `deviceType` is, but is less generic than `source`.

Alternatively, it's possible to add more clarity by giving this param its own type:

```suggestion
	GetHistory(ctx context.Context, user iam.User, order config.LaunchOrder, device DeviceType) (JobHistory, error)
```

```go
// DeviceType is a type that distinguishes between virtual and real devices.
type DeviceType string

const (
	VirtualDevice DeviceType = "vdc"
	RealDevice DeviceType = "rdc"
)
```

Reply:
I mainly used `source` as it was the terminology used by `insights` API (as it can also be `api` as a possible source).
I'm not really convinced by `device` as VDC is not a device per se. I'll think to a better alternative.

Reply:
After thinking a lot I went to the `GetHistory()` invoking both `vdc`and `rdc`  when being called.
The slight delay in execution will be compensated by the more precise response it provides. In the case that is really an issue for customers, we may do it differently.

Meanwhile, this is a more clear and maintainable implementation.